### PR TITLE
[#455] Add checkout repository step to the workflow

### LINKED ISSUE #455: Fix: Publish Wiki Workflow cannot publish wiki doc successfully
## Issue

We don't have checkout the repository step in our workflow from [this discussion](https://nimble-co.slack.com/archives/CSQG2QXFX/p1680670636485339?thread_ts=1680669140.429079&cid=CSQG2QXFX). So the workflow cannot publish the wiki doc after merging to the main branch.

## Expected

The workflow can publish wiki documents successfully.

## Steps to reproduce

1. Merge the wiki markdown page to main branch on the repository.
2. Let the workflow runs.
3. It will return success, but it does not publish the document on wiki page.

### PR DESCRIPTION
**Pull request by lydiasama**
- Closes #455 

## What happened 👀

As discussed in [this thread](https://nimble-co.slack.com/archives/CSQG2QXFX/p1680670636485339?thread_ts=1680669140.429079&cid=CSQG2QXFX):
- Added `Checkout the repository` step to our `publish-wiki-workflow`

## Insight 📝

- The Github wiki repo is using `master` branch by default (Check out [here](https://github.com/orgs/community/discussions/48537)), not `main` as our repository. So in this case, we need to push the wiki content to the `master` branch as the script does correctly. ✅ 

## Proof Of Work 📹

After running the workflow with additional steps, Wiki page has been updated:

Screenshot 2566-04-05 at 12 43 20 PM

### CODE REVIEW

**Review by AVI5HEK — APPROVED** [review]

File: .github/workflows/publish_docs_to_wiki.yml [hunk 1]
```diff
@@ -13,6 +13,16 @@ jobs:
     runs-on: ubuntu-latest
     timeout-minutes: 30
     steps:
+      - name: Cancel previous runs
+        uses: styfle/cancel-workflow-action@0.8.0
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Checkout the repository
+        uses: actions/checkout@v2.3.2
```
Comment (chornerman) [inline .github/workflows/publish_docs_to_wiki.yml:22]: Same here: `actions/checkout@v3`

File: .github/workflows/publish_docs_to_wiki.yml [hunk 2]
```diff
@@ -13,6 +13,16 @@ jobs:
     runs-on: ubuntu-latest
     timeout-minutes: 30
     steps:
+      - name: Cancel previous runs
+        uses: styfle/cancel-workflow-action@0.8.0
```
Comment (chornerman) [inline .github/workflows/publish_docs_to_wiki.yml:17]: I think we should use the latest version: `0.11.0` 🙏🏼

File: .github/workflows/publish_docs_to_wiki.yml [hunk 3]
```diff
@@ -13,6 +13,16 @@ jobs:
...
+        with:
+          access_token: ${{ github.token }}
+
+      - name: Checkout the repository
```
Comment (chornerman) [inline .github/workflows/publish_docs_to_wiki.yml:21]: I think we should start the name of each step with a verb 🤔 
```suggestion
      - name: Check out the repository
```

**Review by thiennguyen0196 — APPROVED** [review]
LGTM

File: .github/workflows/publish_docs_to_wiki.yml (hunk 2 above)
Comment (luongvo) [inline .github/workflows/publish_docs_to_wiki.yml:17]: - Should we [use `concurrency` attributes in workflows to ensure only one job is run for each workflow](https://github.com/nimblehq/compass/pull/1228) instead?
- I'm curious whether we need to add this step to other workflows as well: both [template workflows](https://github.com/nimblehq/android-templates/tree/develop/.github/workflows) & [sample workflows](https://github.com/nimblehq/android-templates/tree/develop/.cicdtemplate).

**Review by luongvo — COMMENTED** [review]
The rest lgtm

File: .github/workflows/publish_docs_to_wiki.yml (hunk 2 above)
Comment (luongvo) [reply .github/workflows/publish_docs_to_wiki.yml:17]: @ryan-conway @lydiasama To make it easy, should we `move the update to cancel the workflow to a separate task` and apply it for the rest too?

Comment (ryan-conway) [reply .github/workflows/publish_docs_to_wiki.yml:17]: @luongvo You may be right, I'll remove this step for now and create an issue to use concurrency on our workflows 💪🏻

Comment (ryan-conway) [reply .github/workflows/publish_docs_to_wiki.yml]: Updated in [7ba2728](https://github.com/nimblehq/android-templates/pull/456/commits/7ba27285d6bdfe75d563fb184228444848006a17)

File: .github/workflows/publish_docs_to_wiki.yml (hunk 3 above)
Comment (ryan-conway) [reply .github/workflows/publish_docs_to_wiki.yml]: Updated in [7ba2728](https://github.com/nimblehq/android-templates/pull/456/commits/7ba27285d6bdfe75d563fb184228444848006a17)

File: .github/workflows/publish_docs_to_wiki.yml (hunk 1 above)
Comment (ryan-conway) [reply .github/workflows/publish_docs_to_wiki.yml]: Updated in [f0826b7](https://github.com/nimblehq/android-templates/pull/456/commits/f0826b7946679981358e529de63faa6c39bcc8b9)

**Comment by luongvo** [comment]
@ryan-conway Could you please also update the PR's title & description? 🙏

File: .github/workflows/publish_docs_to_wiki.yml (hunk 2 above)
Comment (ryan-conway) [reply .github/workflows/publish_docs_to_wiki.yml]: Created [chore ticket](https://github.com/nimblehq/android-templates/issues/462) 🙏🏻

**Review by luongvo — APPROVED** [review]

**Review by chornerman — APPROVED** [review]
LGTM 🚀

**Pull request merged**